packages/react/scroll-area/src/ScrollArea.tsx

```diff
@@ -1,183 +1,77 @@
-  overflowX: 'auto',
-  overflowY: 'auto',
-  scrollbarVisibility: 'hover',
-  scrollbarVisibilityRestTimeout: 600,
-  dir: 'ltr',
-  trackClickBehavior: 'relative',
-  unstable_prefersReducedMotion: false,
-} as const;
-
-interface ScrollAreaRefs {
-  buttonLeftRef: React.RefObject<HTMLDivElement>;
-  buttonRightRef: React.RefObject<HTMLDivElement>;
-  buttonUpRef: React.RefObject<HTMLDivElement>;
-  buttonDownRef: React.RefObject<HTMLDivElement>;
-  viewportRef: React.RefObject<HTMLDivElement>;
-  positionRef: React.RefObject<HTMLDivElement>;
-  scrollAreaRef: React.RefObject<HTMLDivElement>;
-  scrollbarYRef: React.RefObject<HTMLDivElement>;
-  scrollbarXRef: React.RefObject<HTMLDivElement>;
-  thumbYRef: React.RefObject<HTMLDivElement>;
-  thumbXRef: React.RefObject<HTMLDivElement>;
-  trackYRef: React.RefObject<HTMLDivElement>;
-  trackXRef: React.RefObject<HTMLDivElement>;
-}
-
-// Keeping refs in a separate context; should be a stable reference throughout the tree
-const [ScrollAreaRefsProvider, useScrollAreaRefs] = createContext<ScrollAreaRefs>(ROOT_NAME);
-
-interface ScrollAreaContextValue {
-  dir?: TextDirection;
-  overflowX: OverflowBehavior;
-  overflowY: OverflowBehavior;
-  prefersReducedMotion: boolean;
-  scrollbarVisibility: ScrollbarVisibility;
-  scrollbarVisibilityRestTimeout: number;
-  trackClickBehavior: TrackClickBehavior;
-  onScroll: React.ComponentProps<'div'>['onScroll'];
-  isHovered: boolean;
-}
-
-const [ScrollAreaProvider, useScrollAreaContext] = createContext<ScrollAreaContextValue>(ROOT_NAME);
-
-const ScrollAreaStateContext = React.createContext<ScrollAreaReducerState>({} as any);
-ScrollAreaStateContext.displayName = 'ScrollAreaStateContext';
-function useScrollAreaStateContext() {
-  return React.useContext(ScrollAreaStateContext);
-}
-
-// We render native scrollbars initially and switch to custom scrollbars after hydration if the
-// user's browser supports the necessary features. Many internal components will return `null` when
-// using native scrollbars, so we keep implementation separate throughout and check support in this
-// context during render.
-const NativeScrollContext = React.createContext<boolean>(true);
-const useNativeScrollArea = () => React.useContext(NativeScrollContext);
+type ScrollAreaElement = React.ElementRef<typeof ScrollArea>;
+type ViewportElement = React.ElementRef<typeof ScrollAreaViewport>;
+type ScrollbarElement = React.ElementRef<typeof ScrollAreaScrollbar>;
+type ThumbElement = React.ElementRef<typeof ScrollAreaThumb>;
+type Direction = 'ltr' | 'rtl';
+type Sizes = {
+  content: number;
+  viewport: number;
+  scrollbar: {
+    size: number;
+    paddingStart: number;
+    paddingEnd: number;
+  };
+};
 
-const [DispatchProvider, useDispatchContext] = createContext<{
-  dispatch: React.Dispatch<ScrollAreaEvent>;
-}>(ROOT_NAME);
+const MAIN_POINTER = 0;
+const INITIAL_SIZES = {
+  content: 0,
+  viewport: 0,
+  scrollbar: { size: 0, paddingStart: 0, paddingEnd: 0 },
+};
```

Comment:
Sure... I've left a `const mainPointer = 0` var inline because I think the variable name helps communicate what it is without the need for a comment. Thoughts?